internal/controller/shared.go

```diff
@@ -43,14 +43,15 @@ const (
 	RolloutChildNoop RolloutChildOperation = "DO_NOTHING"
 )
 
-// makeChildResourceFromRolloutAndUpdateSpecHash makes a new kubernetes.GenericObject based on the given rolloutObj.
-// It returns the child resource object ready to be created and an operation to be performed with the returned object.
+// makeChildResourceFromRolloutAndCalculatepecHash makes a new kubernetes.GenericObject based on the given rolloutObj.
+// It returns the child resource object ready to be created, an operation to be performed with the returned object,
+// and the calculated hash of the raw spec of the rollout to be set in the rollout annotations after succesful deployment.
 // The operations are defined by the RolloutChildOperation constants.
-func makeChildResourceFromRolloutAndUpdateSpecHash(
+func makeChildResourceFromRolloutAndCalculateSpecHash(
```

Comment:
I hate to be anal, but wondering if we can shorten this name? It's pretty long. I can see it's explicit about what it does, which can be nice. But if you have to name it that way, it sort of reveals a violation of the Single Responsibility Principle, that a function should do one thing. I'm not a total stickler for that rule. (I do think a function can do multiple things but still conceptually do one thing in general.)

Reply:
In any case, can we either shorten the name or break it into multiple functions?

Reply:
I initially envisioned a different usage for this function. But after realizing that certain logic needs to be executed independently of the fact of having a creation or update, lead me to streamline this function into a simpler function.